service/worker/esanalyzer/analyzer.go

```diff
@@ -98,18 +108,28 @@ func New(
 	frontendClient frontend.Client,
 	clientBean client.Bean,
 	esClient es.GenericClient,
+	pinotClient pinot.GenericClient,
 	esConfig *config.ElasticSearchConfig,
 	logger log.Logger,
 	tallyScope tally.Scope,
 	resource resource.Resource,
 	domainCache cache.DomainCache,
 	config *Config,
 ) *Analyzer {
+	var mode readMode
+	if esClient != nil {
+		mode = ES
+	} else if pinotClient != nil {
+		mode = Pinot
+	}
```

Comment:
since this prefers ES if both are present:
could this mode be replaced by "just use the non-nil client" and only pass one?

or is there more stuff coming that'll use both sometimes?

Reply:
Yes, it can be. It's just that Pinot and ES has different GenericClients, there'll need a lot efforts to do a refactor.

Reply:
aaah, because esClient is used by other things too, so it's not just for this mode.
alrighty, makes sense 👍